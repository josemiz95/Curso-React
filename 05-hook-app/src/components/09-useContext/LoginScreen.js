import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {user, setUser} = useContext(UserContext);

    const handleLogIn = () => setUser({ id: 123, name:"Josemi" });
    const handleLogOut = () => setUser(null);

    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>
            {
                user === null?
                <button className='btn btn-primary' onClick={handleLogIn}>Login</button>
                :
                <button className='btn btn-warning' onClick={handleLogOut}>LogOut</button>
            }
        </div>
    )
}
