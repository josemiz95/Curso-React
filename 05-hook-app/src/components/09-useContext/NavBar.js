import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">
                <Link to={"/"} className="navbar-brand" href="/">Navbar</Link>
                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav">
                        <li className="nav-item"><NavLink activeClassName="active" className='nav-link' to={"/"}>Home</NavLink></li>
                        <li className="nav-item"><NavLink activeClassName="active" className='nav-link' to={"/about"}>About</NavLink></li>
                        <li className="nav-item"><NavLink activeClassName="active" className='nav-link' to={"/login"}>Login</NavLink></li>
                    </ul>
                </div>
            </div>
        </nav>
    )
}
