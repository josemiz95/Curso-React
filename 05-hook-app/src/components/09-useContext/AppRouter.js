import React from 'react';
import { BrowserRouter, Routes, Route, Navigate } from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <div>
                <NavBar />
                <div className='container'>
                    <Routes>
                        <Route path='/' element={<HomeScreen />}/>
                        <Route exact path='/login' element={<LoginScreen />}/>
                        <Route exact path='/about' element={<AboutScreen />}/>
                        <Route exact path='/about/you' element={<AboutScreen />}/>

                        {/* Default route: */}
                        <Route path='*' element={<Navigate to={"/"} replace/>}/>
                    </Routes>
                </div>
            </div>
        </BrowserRouter>
    )
}
