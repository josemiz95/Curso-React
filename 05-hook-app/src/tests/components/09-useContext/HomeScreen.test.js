import { mount } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('test on <HomeScreen />', () => {
	const user = {name: "Josemi", email: "josemiz95"}

	const wrapper = mount(
		<UserContext.Provider value={user}>
			<HomeScreen />
		</UserContext.Provider>
	);

	test('should show the component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});