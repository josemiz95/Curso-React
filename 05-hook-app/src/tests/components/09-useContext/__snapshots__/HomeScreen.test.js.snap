// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <HomeScreen /> should show the component 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre />
</div>
`;
